File: InventoryManagementServices/ReactApp/inventory-management-react/src/app/models/movies/movieSearchResults.ts
import { UndoRounded } from "@mui/icons-material";

export interface MovieSearchResults
{
    id?: string;
    title: string;
    description: string;
    movieRating: string;
    moveGenre: string;
    hours?: string;
    minutes?: string;
    boxOffice?: string;

}


export class MovieSearchResults implements MovieSearchResults {
    constructor(init?: MovieSearchResultsFormValues) {
        Object.assign(this, init);
    }
}

  export class MovieSearchResultsFormValues {
    id?: string = undefined;
    title: string = '';
    description: string= '';
    movieRating: string= '';
    moveGenre: string = '';
    hours?: string = undefined;
    minutes?: string = undefined;
    boxOffice?: string = undefined;

    constructor(movie?: MovieSearchResultsFormValues) {
      if (movie) {
        this.id = movie.id
        this.title = movie.title;
        this.description = movie.description;
        this.movieRating = movie.movieRating;
        this.moveGenre = movie.moveGenre;
        this.hours = movie.hours;
        this.minutes = movie.minutes;
        this.boxOffice = movie.boxOffice;
      }
    }
  }

File: InventoryManagementServices/Services/Movie/Movie.Infrastructure/Entities/MovieSearchResults.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Movie.Infrastructure.Entities
{
    public class MovieSearchResults
    {
        public Guid? Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime? ReleaseDate { get; set; }
        public int? Hours { get; set; }
        public int? Minutes { get; set; }
        public decimal? BoxOffice { get; set; }
        public string MovieGenre { get; set; }
        public string MovieRating { get; set; }
    }
}
